### PR TITLE
OpenLDAP Syncrepl support

### PR DESCRIPTION
**Pull request by tpretz**
It can be useful to be able to interact with the OpenLDAP replication standard, both as a client and as a server.

This PR will add support for the various controls, and messages that are required

### CODE REVIEW

File: lib/controls/sync_done_control.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
+// Copyright 2011 Mark Cavage, Inc.  All rights reserved.
+
```
Comment (jsumners) [inline lib/controls/sync_done_control.js:2]: These new files should not start with this outdated copyright notice. They should also start with `'use strict'`.
```suggestion
'use strict'

```

**Comment by UziTech** [comment]
Looks like this just needs some tests.

**Comment by clayrisser** [comment]
Also, how would I use this? Is there a basic example?

**Comment by UziTech** [comment]
This just needs some tests in order to be merged in. @tpretz are you still working on this?

**Comment by clayrisser** [comment]
@tpretz  do you have an example on how to use this?

**Comment by jsumners** [comment]
@clayrisser unfortunately, I think this PR is abandoned. You can take it up by following https://gist.github.com/jsumners/461ef7a64545108635cc437fde112721

**Comment by tpretz** [comment]
unfortunately i have not had time recently to work on the project that was pushing these additions

When i do get back to it i'll raise again, or if anyone else wants to look at / enhance my source branch will remain.

**Comment by jsumners** [comment]
⚠️ This issue has been locked due to age. If you have encountered a recent
problem that seems to be covered by this issue, please open a new issue.

Please include a [minimal reproducible example](https://en.wikipedia.org/wiki/Minimal_reproducible_example)
when opening a new issue.

**Pull request closed without merging**